src/types/submission.types.ts

```diff
@@ -0,0 +1,31 @@
+export type Submission = {
+  id?: number
+  userId: number
+  assignmentId: number
+  courseId: number
+  type: SubmissionType
+  content: string
+  submitterIp?: string
+
+  // submitterId can be different from userId if a submission was created on behalf of the student (regrade)
+  // originalSubmissionId is only used on regrades
+  submittedBy: number | null
```

Comment:
Will submittedBy ever be null? In the API PR the userId is grabbed from the POST request which should never be null

Reply:
Oh wait, I'm dumb. It will be null on the client side.